### ISSUE TITLE
Feature Requests for Array-Widget

### ISSUE THREAD

**Issue by Amadrath** [ControlSystemStudio/phoebus#1412]
Right now the array widget seems to be static. You add a widget to the array and have to change its size to fit the number of copies you want, which are generated then and there. It would be better, if the copies were generated on runtime. The frame size of the array widget could optionally either be static, with a scroll bar appearing, if the number of widgets surpasses the available space in the frame, or dynamic.
It would also be nice to have a macro, which holds the index of the array, to use i.e. for labeling. (Like $(pv_array_n) = 0 for first widget in array, 1 for second, and so on.)

**Comment by exzombie** [comment]
Indeed, having to instantiate the widgets manually is a major pain point, defeating most of the purpose of this widget. Especially because the array example shipped with Phoebus claims that "At runtime, it creates one copy of the child widget for each array element", which is not currently true.

Another big issue is the orientation of the widget. It's nice that it can infer the proper orientation based on the orientation of the widgets within. But widgets without the "Horizontal" property, like LEDs, are always stacked vertically, which is not always appropriate.
